pkg/feature/feature.go

```diff
@@ -59,52 +70,60 @@ func (f *Feature) Apply() error {
 		multiErr = multierror.Append(multiErr, precondition(f))
 	}
 
-	if multiErr.ErrorOrNil() != nil {
-		return multiErr.ErrorOrNil()
+	preconditionsErr := multiErr.ErrorOrNil()
+	if preconditionsErr != nil {
+		f.Log.Error(preconditionsErr, "failed ensuring preconditions are met")
+		return preconditionsErr
 	}
 
 	// Load necessary data
 	for _, loader := range f.loaders {
 		multiErr = multierror.Append(multiErr, loader(f))
 	}
-	if multiErr.ErrorOrNil() != nil {
-		return multiErr.ErrorOrNil()
+
+	dataLoadErr := multiErr.ErrorOrNil()
+	if dataLoadErr != nil {
+		f.Log.Error(dataLoadErr, "failed loading template data")
+		return dataLoadErr
 	}
 
 	// create or update resources
 	for _, resource := range f.resources {
 		if err := resource(f); err != nil {
-			return err
+			f.Log.Error(err, "failed creating a resource")
```

Comment:
log and return?

Reply:
Not sure what you mean. One or the other? I'm doing both now

Reply:
Yeah, handle or "throw", not both. :)

Reply:
The idea was to leverage the fact that at this point we know the phase of where the error happen so we can add a bit more context to the log. But with #801 coming, I think we can handle it in `defer` there. Removed logging in 3cb69b3